### DISCUSSION TITLE
setQueryData event

### DISCUSSION THREAD

**Discussion by fonderkin** [TanStack/query#4075] [Ideas] [answered]
Hello!

It would be desirable that after setQueryData, isLoading is updated or a new isUpdateChache parameter is introduced.

Motivation:
A table with data, when changing the table element asynchronously after setting setQueryData, assign the loading status to the table so that the user does not see the change

https://user-images.githubusercontent.com/16304172/186428741-5d279ac0-29e7-40f8-9f27-c94a49dcfbd4.mp4

.

**Comment by TkDodo** [comment, marked as answer]
if you are doing refetches, the query will have `isFetching:true`. If you are updating with `setQueryData`, the updates are synchronous, so they will be reflected instantly. Not sure what you are doing without seeing the code
